index.go

```diff
@@ -331,6 +331,15 @@ func (idx *faissIndex) Reconstruct(key int64) (recons []float32, err error) {
 func (idx *faissIndex) ReconstructBatch(keys []int64, recons []float32) ([]float32, error) {
 	var err error
 	n := int64(len(keys))
+	if recons == nil {
+		recons = make([]float32, n*int64(idx.D()))
+	}
+
+	// exit in case of invalid input
+	if n == 0 || len(recons) != int(n)*idx.D() {
```

Comment:
Perhaps the `n == 0` check should be moved up before the recons alloc - line 333.5.

Reply:
technically speaking this is still the first validation check, so i dont think it would cause issues if we keep it over here. (just that moving the n == 0 check will have three separate if conditions which wouldn't look good i guess)

the condition at 334 doesn't exit the function, since its for when some user doesn't know what exactly to give for recons (which is like a prealloc) - in which case the function explicitly allocs the slice.